services/worker/src/worker/job_runners/split/duckdb_index.py

```diff
@@ -0,0 +1,280 @@
+    return {
+        "dataset": dataset,
+        "config": config,
+        "split": split,
+        "url": hf_hub_url(
+            repo_id=dataset,
+            filename=repo_file.rfilename,
+            hf_endpoint=hf_endpoint,
+            revision=target_revision,
+            url_template=url_template,
+        ),
+        "filename": Path(repo_file.rfilename).name,
+        "size": repo_file.size,
+    }
+
+
+def compute_index_rows(
+    dataset: str,
+    config: str,
+    split: str,
+    duckdb_index_directory: StrPath,
+    target_revision: str,
+    hf_endpoint: str,
+    commit_message: str,
+    url_template: str,
+    hf_token: Optional[str],
+    committer_hf_token: Optional[str],
+) -> IndexRowsResponse:
+    logging.info(f"get split-duckdb-index for dataset={dataset} config={config} split={split}")
+
+    # validate split
+    split_names_best_response = get_previous_step_or_raise(
+        kinds=["config-split-names-from-streaming", "config-split-names-from-info"], dataset=dataset, config=config
+    )
+    try:
+        splits_content = split_names_best_response.response["content"]["splits"]
+    except Exception as e:
+        raise PreviousStepFormatError("Previous step did not return the expected content.", e) from e
+
+    if split not in [split_item["split"] for split_item in splits_content]:
+        raise SplitNotFoundError(f"The split '{split}' does not exist for the config '{config}' of the dataset.")
+
+    # get parquet content
+    config_parquet_best_response = get_previous_step_or_raise(kinds=["config-parquet"], dataset=dataset, config=config)
+    try:
+        parquet_files_content = config_parquet_best_response.response["content"]["parquet_files"]
+        parquet_file_items: List[ParquetFileItem] = [
+            parquet_file_item
+            for parquet_file_item in parquet_files_content
+            if parquet_file_item["config"] == config and parquet_file_item["split"] == split
+        ]
+        if not parquet_file_items:
+            raise ParquetResponseEmptyError("No parquet files found.")
+    except Exception as e:
+        raise PreviousStepFormatError("Previous step did not return the expected content.") from e
+
+    fs = HTTPFileSystem()
+    parquet_urls = [parquet_file_item["url"] for parquet_file_item in parquet_file_items]
+    desc = f"{dataset}/{config}"
+    try:
+        parquet_files: List[ParquetFile] = thread_map(
+            partial(get_parquet_file, fs=fs, hf_token=hf_token), parquet_urls, desc=desc, unit="pq", disable=True
+        )
+    except Exception as e:
+        raise FileSystemError(f"Could not read the parquet files: {e}") from e
+
+    # get the features
+    features = Features.from_arrow_schema(parquet_files[0].schema.to_arrow_schema())
+
+    # look for string columns using the first rows
+    string_columns = [column for column, feature in features.items() if STRING_FEATURE_DTYPE in str(feature)]
```

Comment:
and in this case, if only first parquet file is taken, is it needed to get all of them (in line 101)?

for visibility: @lhoestq approved that "features" field should always exist in "dataset_info" and if it's not - it's a bug. So it should be safe to take features from there, instead of reading parquet schema from files directly.

Reply:
Thanks @polinaeterna ! I will do the change to read from dataset_info instead

Reply:
I changed the parent to "config-parquet-and-info" and now, it gets features from cache and no need to load from parquet file  schema.